src/app/forms/validators/date/date.validators.ts

```diff
@@ -6,9 +6,24 @@ export class DateValidators {
       if (!control.value) {
         return null;
       }
-
       const [d, t] = (control.value as string).split('T');
       const [year, month, day] = d.split('-');
+
+      if (!day) {
+        return null;
+      }
+
+      if (!month) {
+        return null;
+      }
+      if (!year) {
+        return null;
+      }
+
+      if (year.length < 4) {
+        return null;
+      }
+
```

Comment:
I don't think adding these is the way to go. the responsibility of the validator is the validate a date, it should say if a given input is a valid date or not. Handling the displaying or not of errors should not be the validator's responsibility

Reply:
> I don't think adding these if statements is the way to go. the responsibility of the validator is the validate a date, it should say if a given input is a valid date or not. Handling the displaying or not of errors should not be the validator's responsibility

Where would you suggest this logic to happen?